include/nccl_ofi_ep_addr_list.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All rights reserved.
+ */
+
+#ifndef NCCL_OFI_EP_ADDR_LIST_H
+#define NCCL_OFI_EP_ADDR_LIST_H
+
+#include "nccl_ofi.h"
```

Comment:
For `nccl_net_ofi_ep_t`.

I made the ep_addr_list take in pointers of this type for the endpoint to have some semblance of type checking (I'm trying to minimize use of void pointers). But I'm not ecstatic about using `nccl_net_ofi_ep_t`, because it suggests this structure is doing something specific to endpoints, whereas in reality it's treating the endpoint as an opaque pointer.

Suggestions welcome.

Reply:
if it's just an opaque pointer, use a void*.  But also, since you don't actually look at the endpoint, you can just declare `struct nccl_net_ofi_ep_t;' in this file and not bring in all of nccl_ofi.h.

Reply:
I took your second suggestion.